pkg/controller/storagecluster/externalstoragecluster_controller.go

```diff
@@ -63,6 +63,27 @@ func (r *ReconcileStorageCluster) ReconcileExternalStorageCluster(sc *ocsv1.Stor
 			return reconcile.Result{}, err
 		}
 	}
+
+	// Handle CephCluster resource status
+	if found.Status.State == "" {
```

Comment:
found.Status.Phase == ""

State will be changed to Phase on rook 1.3 and future OCS versions will be using Phase variable.

Reply:
@nizamial09 could you please provide some reference for this change ?

Reply:
https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/types.go#L145

So for rook:master the `Phase` is used. Any build starting from rook v1.3 will starts to use `Phase` instead of `State`. So I guess its better to start using `Phase` here also.

Reply:
@umangachapagain ^

Reply:
Yes. Use rook:master.

Reply:
@nizamial09 Don't we set both Phase and State in master? I thought we needed to set State as well to avoid breaking changes. For example, there may by others than OCS that are using the State and would be surprised by the upgrade to 1.3.
OCS likely doesn't want to use the `rook/ceph:master` tag as it will be a bit unstable. Perhaps a particular master tag could be used and updated periodically if we need to move to master though.

Reply:
I see your concern. My PR will completely eliminate the use of State from the master. But you are correct. We need to be doing it slowly. I will make another PR which will bring back the State for now and in future we can slowly deprecate it's use. @travisn

Reply:
Made the changes from State to Phase